packages/wallet/src/redux/protocols/responding/__tests__/reducer.test.ts

```diff
@@ -0,0 +1,211 @@
+import * as scenarios from './scenarios';
+import { initialize, respondingReducer } from '../reducer';
+
+import * as states from '../state';
+import { Commitment } from 'fmg-core/lib/commitment';
+import * as TransactionGenerator from '../../../../utils/transaction-generator';
+
+// Mocks
+const mockTransaction = { to: '0xabc' };
+const createRespondWithMoveMock = jest.fn().mockReturnValue(mockTransaction);
+const refuteMock = jest.fn().mockReturnValue(mockTransaction);
+Object.defineProperty(TransactionGenerator, 'createRespondWithMoveTransaction', {
+  value: createRespondWithMoveMock,
+});
+Object.defineProperty(TransactionGenerator, 'createRefuteTransaction', {
+  value: refuteMock,
+});
+
+// helpers
+const itTransitionsToFailure = (
+  result: { protocolState: states.RespondingState },
+  failure: states.Failure,
+) => {
+  it(`transitions to failure with reason ${failure.reason}`, () => {
+    expect(result.protocolState).toMatchObject(failure);
+  });
+};
+
+const itCallsRespondWithMoveWith = (challengeCommitment: Commitment) => {
+  it('calls respond with move with the correct commitment', () => {
+    expect(createRespondWithMoveMock).toHaveBeenCalledWith(
+      challengeCommitment,
+      jasmine.any(String),
+    );
+  });
+};
+
+const itCallsRefuteWith = (refuteCommitment: Commitment) => {
+  it('calls refute with the correct commitment', () => {
+    expect(refuteMock).toHaveBeenCalledWith(refuteCommitment, jasmine.any(String));
+  });
+};
+
+const itTransitionsTo = (result: { protocolState: states.RespondingState }, type: string) => {
+  it(`transitions to ${type}`, () => {
+    expect(result.protocolState.type).toEqual(type);
+  });
+};
+
+const itSetsChallengeCommitment = (
+  result: { protocolState: states.RespondingState },
+  commitment: Commitment,
+) => {
+  it('sets the correct challenge commitment', () => {
+    expect((result.protocolState as states.WaitForApproval).challengeCommitment).toMatchObject(
+      commitment,
+    );
+  });
+};
+
+describe('respond with existing move happy-path scenario', () => {
+  const scenario = scenarios.respondWithExistingCommitmentHappyPath;
+  const { sharedData, processId } = scenario;
+
+  describe('when initializing', () => {
+    const { challengeCommitment } = scenario;
+    const result = initialize(processId, sharedData, challengeCommitment);
+
+    itTransitionsTo(result, states.WAIT_FOR_APPROVAL);
+    itSetsChallengeCommitment(result, scenario.challengeCommitment);
+  });
+
+  describe(`when in ${states.WAIT_FOR_APPROVAL}`, () => {
```

Comment:
I would probably go further and just have `describe('when in WaitForApproval', () => {`. I think that's marginally easier to read and you can then directly match with jest's output on the command line.